flexmeasures/api/v3_0/tests/test_assets_api_fresh_db.py

```diff
@@ -73,3 +74,114 @@ def test_delete_an_asset(client, setup_api_fresh_test_data, requesting_user, db)
         select(GenericAsset).filter_by(id=existing_asset_id)
     ).scalar_one_or_none()
     assert deleted_asset is None
+
+
+@pytest.mark.parametrize(
+    "requesting_user, sensor_index, data_unit, data_resolution, price, expected_event_values, expected_status",
+    [
+        (
+            "test_prosumer_user_2@seita.nl",
+            1,
+            "m/s",
+            "1h",
+            45.3,
+            None,
+            422,
+        ),  # this sensor has unit=kW
+        (
+            "test_prosumer_user_2@seita.nl",
+            2,
+            "kWh",
+            "1h",
+            45.3,
+            45.3,
+            200,
+        ),  # this sensor has unit=kWh
+        (
+            "test_prosumer_user_2@seita.nl",
+            0,
+            "kWh",
+            "1h",
+            45.3,
+            0.0453,
+            200,
+        ),  # this sensor has unit=MW
+        (
+            "test_prosumer_user_2@seita.nl",
+            1,
+            "MW",
+            "1h",
+            2,
+            2000,
+            200,
+        ),  # this sensor has unit=kW
+        (
+            "test_prosumer_user_2@seita.nl",
+            1,
+            "kWh",
+            "30min",
+            10,
+            20,
+            200,
+        ),  # this sensor has unit=kW
+    ],
+    indirect=["requesting_user"],
+)
+def test_auth_upload_sensor_data_with_distinct_units(  # TODO: remove auth prefix from function name
```

Comment:
Why is this test in the module `test_asset_api`(again)? 

It tests the sensor API, so move it there.

And do the renaming (remove "_auth"), I even made a clickable suggestion for that around two reviews ago.